test/models/network.js

```diff
@@ -147,4 +147,194 @@ describe("Network", function() {
+			const chan2 = new Chan({name: "#THELOUNGE"});
+
+			const network = new Network({
+				channels: [
+					chan1,
+					chan2,
+				],
+			});
+
+			const newUser = new Chan({name: "mcinkay", type: Chan.Type.QUERY});
+			network.addChannel(newUser);
+
+			expect(network.channels[1]).to.equal(chan1);
+			expect(network.channels[2]).to.equal(chan2);
+			expect(network.channels[3]).to.equal(newUser);
+		});
+
+		it("should sort users alphabetically", function() {
+			const chan1 = new Chan({name: "#abc"});
+			const chan2 = new Chan({name: "#THELOUNGE"});
+			const user1 = new Chan({name: "astorije", type: Chan.Type.QUERY});
+			const user2 = new Chan({name: "xpaw", type: Chan.Type.QUERY});
+
+			const network = new Network({
+				channels: [
+					chan1,
+					chan2,
+					user1,
+					user2,
+				],
+			});
+
+			const newUser = new Chan({name: "mcinkay", type: Chan.Type.QUERY});
+			network.addChannel(newUser);
+
+			expect(network.channels[1]).to.equal(chan1);
+			expect(network.channels[2]).to.equal(chan2);
+			expect(network.channels[3]).to.equal(user1);
+			expect(network.channels[4]).to.equal(newUser);
+			expect(network.channels[5]).to.equal(user2);
+		});
+
+		it("should not sort special channels", function() {
+			const chan1 = new Chan({name: "#abc"});
+			const chan2 = new Chan({name: "#THELOUNGE"});
+			const user1 = new Chan({name: "astorije", type: Chan.Type.QUERY});
+			const user2 = new Chan({name: "xpaw", type: Chan.Type.QUERY});
+
+			const network = new Network({
+				channels: [
+					chan1,
+					chan2,
+					user1,
+					user2,
+				],
+			});
+
+			const newBanlist = new Chan({name: "Banlist for #THELOUNGE", type: Chan.Type.SPECIAL});
+			network.addChannel(newBanlist);
+
+			expect(network.channels[1]).to.equal(chan1);
+			expect(network.channels[2]).to.equal(chan2);
+			expect(network.channels[3]).to.equal(user1);
+			expect(network.channels[4]).to.equal(user2);
+			expect(network.channels[5]).to.equal(newBanlist);
+		});
+
+		it("should not compare against special channels", function() {
+			const chan1 = new Chan({name: "#abc"});
+			const chan2 = new Chan({name: "#THELOUNGE"});
+			const user1 = new Chan({name: "astorije", type: Chan.Type.QUERY});
+
+			const network = new Network({
+				channels: [
+					chan1,
+					chan2,
+					user1,
+				],
+			});
+
+			const newBanlist = new Chan({name: "Banlist for #THELOUNGE", type: Chan.Type.SPECIAL});
+			network.addChannel(newBanlist);
+			const newUser = new Chan({name: "mcinkay", type: Chan.Type.QUERY});
+			network.addChannel(newUser);
+
+			expect(network.channels[1]).to.equal(chan1);
+			expect(network.channels[2]).to.equal(chan2);
+			expect(network.channels[3]).to.equal(user1);
+			expect(network.channels[4]).to.equal(newUser);
+			expect(network.channels[5]).to.equal(newBanlist);
+		});
+
+		it("should insert before first special channel", function() {
+			const banlist = new Chan({name: "Banlist for #THELOUNGE", type: Chan.Type.SPECIAL});
```

Comment:
Until we have a better UI for banlist/channel list, it would be nice if banlists would open right after their corresponding channel, and channel list open at the top of the channel list (well right below server window).

That's an entirely different PR, but just wanted to point that out in case you were thinking about a follow-up PR or something.